src/chapters/objects-and-math/codingwithobjects.rst

```diff
@@ -0,0 +1,163 @@
+      :linenos:
+
+      let tortoiseTwo = {
+          species: "Galapagos Tortoise",
+          name: "Patricia",
+          weight: 800,
+          age: 85,
+          diet: ["pumpkins", "lettuce", "cabbage"],
+          sign: function() {
+              return this.name + " is a " + this.species;
+          }
+       };
+
+   Because Pete and Patricia are members of the same species, are the same age, and have the same diet, you might notice that many of their properties are equal, but some are not.
+   Pete weighs more than Patricia and of course, they have different names!
+
+   For this example, we will only keep the ``species`` and ``diet`` properties.
+
+   .. sourcecode:: js
+      :linenos:
+
+      let tortoiseOne = {
+          species: "Galapagos Tortoise",
+          diet: ["pumpkins", "lettuce", "cabbage"]
+       };
+
+       let tortoiseTwo = {
+          species: "Galapagos Tortoise",
+          diet: ["pumpkins", "lettuce", "cabbage"],
+       };
+
+       console.log(tortoiseOne === tortoiseTwo);
+
+   **Console Output**
+
+   ::
+
+      False
+
+   The objects contain properties that have the same keys and equal values.
+   However, the output is ``False``.
+
+Even though ``tortoiseOne`` and ``tortoiseTwo`` have the same keys and values, they are stored in separate locations in memory.
+This means that even though you can compare the properties in different objects for equality, you cannot compare the objects themselves for equality.
+
+Iterating Through Objects
+-------------------------
+
+We can iterate through all of the values in an object, much like we would do with an array.
+We will use a ``for`` loop to do just that!
+
+For objects, we use a different ``for`` loop structure.
+``for..in`` loops are specifically designed to easily loop through the properties in an object.
+Each iteration of the loop accesses a key in the object.
+The loop stops once it has accessed every property.
+
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+      :linenos:
+
+      let giraffe = {
+        species: "Reticulated Giraffe",
+        name: "Cynthia",
+        weight: 1500,
+        age: 15,
+        diet: "leaves"
+      };
+
+      for (item in giraffe) {
+         console.log(item, giraffe[item]);
+      }
+
+   **Console Output**
+
+   ::
+
+      species, Reticulated Giraffe
+      name, Cynthia
+      weight, 1500
+      age, 15
+      diet, leaves
+
+   In this example, ``item`` is a variable that holds the string for each key. It is updated with each iteration of the loop.
+
+.. note::
+
+   Inside a ``for..in`` loop, we can only use bracket syntax to access the property values.
+
+.. admonition:: Try It
+
+   Write a ``for..in`` loop to print to the console the values in the ``tortoiseOne`` object. `TRY IT! <https://repl.it/@launchcode/forinLoop/>`_
+
+
+Objects and Functions
+---------------------
+
+Programmers can pass objects as in the input of a function or use an object as the return value of the function.
```

Comment:
Typo and comma corrections: "Programmers can pass an object as the input to a function, or use an object as the return value of the function."

Reply:
A comma doesn't actually come before the "or" because "or" is followed by a dependent clause. I will think about some examples.

Reply:
Thanks for the punctuation refresher.  :-)